skyportal/plot.py

```diff
@@ -1398,13 +1387,78 @@ def spectroscopy_plot(obj_id, user, spec_id=None, width=600, device="browser"):
     if len(spectra) == 0:
         return None, None, None
 
+    # sort out the size of the plot
+
+    layouts = []
+    spectra_by_type = collections.defaultdict(list)
+
+    for s in spectra:
+        spectra_by_type[s.type].append(s)
+
+    # sort the dictionary to be ordered according to the given list
+    def sorted_dict(d, key_list):
+        return {key: d[key] for key in key_list if key in d}
+
+    # we want the tabs to appear in the order they're defined in the config:
+    spectra_by_type = sorted_dict(spectra_by_type, ALLOWED_SPECTRUM_TYPES)
+
+    # go over the types in the order they're defined in the config
+    for spec_type in ALLOWED_SPECTRUM_TYPES:
+
+        if len(spectra_by_type[spec_type]) == 0:
+            continue
+
+        layouts.append(
+            make_spectrum_layout(obj, spectra_by_type[spec_type], user, device, width)
+        )
+
+    if len(layouts) > 1:
+        panels = []
+        spectrum_types = [s for s in spectra_by_type]
```

Comment:
This assumes that only valid spectrum types are given?

I think what you want then is to replace the above with:

```
# Filter out spectra with types that are not allowed
spectra_by_type = {key: spectra_by_type[key] for key in ALLOWED_SPECTRUM_TYPES if key in spectra_by_type}

for spec_type in spectra_by_type:
    layouts.append((spec_type, make_spectrum_layout(...))

if layouts:
    panels = []
    for (stype, layout) in layouts:
        panels.append(Panel(child=layout, title=stype))
```

Reply:
I think that is the same thing as what I'm doing with the function you suggested before. And also, the spectra must be one of the allowed spectrum types (this is enforced by the database).

Reply:
If that's the case, you can simply:

```
for spec_type in spectra_by_type:
    ...
```

And remove the `len(...) == 0` check.

Reply:
I think I just did that on the latest commit.